// _newsletter/newsletter.js
// STACK
var h       = require('virtual-hyperscript-hook')(require('virtual-dom/h'))
// CUSTOM
var styling = require('_styling')
var s

module.exports = NewsletterComponent

function NewsletterComponent (state) {
  s = s ? s : styling(`
      .newsletter                   {
        display                     : flex;
        position                    : relative;
        flex-direction              : row;
        justify-content             : center;
        align-items                 : center;
        background-color            : ${state.theme.colors.white};
        height                      : 40px;
        width                       : 200px;
        border-radius               : 20px;
        text-transform              : uppercase;
        color                       :  ${state.theme.colors.darkergrey};
        font-size                   :  ${state.theme.font.small+'px'};
        font-family                 :  ${state.theme.font.avenirRoman};
        text-decoration             : none;
      }
      .newsletter--hover            {
      color                       :  ${state.theme.colors.red};
      }
    }
  `)
  return function Newsletter (state) {
    return link({
      state: state,
      namespace: 'newsletter',
      name: 'newsletter',
      text: 'Newsletter',
      url: '/#newsletter',
      csshover: '.newsletter--hover',
      cssnormal: '.newsletter'
    })
  }
}
// SUBTEMPLATES
function link ({state, namespace, name, text, url, csshover, cssnormal} = {}) {
  if (!state.ui[namespace]) { state.ui[namespace] = {} }
  return h('a', {
    className     : state.ui[namespace][name] ?
      s(cssnormal,csshover) : s(cssnormal),
    onmouseenter (event) {
      if (!state.ui[namespace][name]) {
        state.ui[namespace][name] = true
        state.action('#! hover: '+name, state)
      }
    },
    onmouseleave (event) {
      if (state.ui[namespace][name]) {
        state.ui[namespace][name] = false
        state.action('#! unhover: '+name, state)
      }
    },
    href          : url
  }, text)
}
// router.addRoute('/apply', function (m) {
//   return h('form', { method: 'POST', action: '/apply' }, [
//     h('div', [
//       'name',
//       h('input', { type: 'text', name: 'name' }),
//     ]),
//     h('input', { type: 'submit', value: 'apply' })
//   ])
// })
// function post (fn) {
//     return function (req, res, params) {
//         if (req.method !== 'POST') {
//             res.statusCode = 400
//             res.end('not a POST\n')
//         }
//         body(req, res, function onbody (err, pvars) {
//             fn(req, res, xtend(pvars, params))
//         })
//     }
// }
// router.addRoute('/submit', post(function (req, res) {
//     console.log(params)
//     res.end('ok!\n')
// }))
