// _header/header.js
// STACK
var h         = require('virtual-hyperscript-hook')(require('virtual-dom/h'))
// CUSTOM
var styling   = require('_styling')
var s

var isretina = require('is-retina')

module.exports = HeaderComponent

function HeaderComponent (state) {
  s = s ? s : styling(`
    .header {
      align-items         : center;
      align-self          : center;
      display             : flex;
      justify-content     : space-between;
      min-width           : ${state.theme.header.width+'px'};
      max-width           : ${state.theme.header.width+'px'};
      flex-wrap           : nowrap;
    }
    .logo {
      display             : flex;
      align-self          : center;
      margin              : 30px;
    }
    /*
    .logo {
      background-image    : url("assets/redlabR.svg");
      background-size     : 45px 45px;
      background-repeat   : no-repeat;
      color               : ${state.theme.colors.grey};
      display             : flex;
      flex-wrap           : nowrap;
      font-family         : ${state.theme.font.hnul};
      font-size           : ${state.theme.font.sizeA+'pt'};
      font-weight         : 600;
      height              : 45px;
      margin-top          : 20px;
      margin-bottom       : 20px;
      padding-left        : 50px;
      position            : relative;
      justify-content     : center;
      align-items         : center;
      text-decoration     : none;
    }
    */
    .menu {
      display             : flex;
      flex-wrap           : nowrap;
    }
    .item {
      display             : flex;
      color               : ${state.theme.colors.darkgrey};
      font-family         : ${state.theme.font.avenir.roman.family};
      font-size           : ${state.theme.font.avenir.roman.size+'pt'};
      font-weight         : 600;
      margin-left         : 50px;
      text-decoration     : none;
      text-transform      : uppercase;
      padding-top         : 10px;
      padding-bottom      : 10px;
      position            : relative;
      white-space         : nowrap;
    }
    .item--hover {
      color               : ${state.theme.colors.red};
    }
    .submenu {
      background-color    : ${state.theme.colors.black};
      display             : flex;
      flex-direction      : column;
      flex-wrap           : nowrap;
      justify-content     : center;
      left                : 0px;
      min-width           : 100%;
      overflow            : hidden;
      position            : absolute;
      top                 :  ${17+parseInt(state.theme.font.avenir.small.size)+'pt'};
      z-index             : 1;
    }
    .submenu--visible {
      display             : flex;
    }
    .submenu--hidden {
      display             : none;
    }
    .subitem {
      border-top-style    : solid;
      border-top-width    : 3px;
      border-top-color    : ${state.theme.colors.black};
      border-bottom-style : solid;
      border-bottom-width : 3px;
      border-bottom-color : ${state.theme.colors.black};
      color               : ${state.theme.colors.white};
      display             : flex;
      font-family         : ${state.theme.font.hnul};
      font-size           : 15px;
      font-weight         : 100;
      text-decoration     : none;
      text-transform      : uppercase;
      padding-bottom      : 20px;
      padding-left        : 10px;
      padding-right       : 10px;
      padding-top         : 20px;
      white-space         : nowrap;
    }
    .subitem--hover {
      border-top-color    : ${state.theme.colors.red};
      border-bottom-color : ${state.theme.colors.red};
    }
  `)
  return function Header (state) {
    // DEFAULTS
    if (!state.ui.submenu) { state.ui.submenu = {} }
    // TEMPLATE
    return (
      h('div', { className: s('.header') }, [
        h('a', { className: s('.logo'), href: '/'}, [
          // state.data.title
          h('img', { src: state.data.logo[isretina()?'retina':'normal'] })
        ]),
        h('div', { className: s('.menu') }, [
          h('a', {
            className     : state.ui.submenu['products'] ?
              s('.item','.item--hover') : s('.item'),
            onmouseenter (event) {
              if (!state.ui.submenu['products']) {
                state.ui.submenu['products'] = true
                state.action('#! open submenu: products', state)
              }
            },
            onmouseleave (event) {
              if (state.ui.submenu['products']) {
                state.ui.submenu['products'] = false
                state.action('#! close submenu: products', state)
              }
            },
            href          : '/#products',
          }, [
            h('span', 'Produkte'),
            submenu(state)
          ]),
          h('span', [
            h('a', {
              className     : state.ui.submenu['about'] ?
                s('.item','.item--hover') : s('.item'),
              onmouseenter () {
                if (!state.ui.submenu['about']) {
                  state.ui.submenu['about'] = true
                  state.action('#! open submenu: about', state)
                }
              },
              onmouseleave () {
                if (state.ui.submenu['about']) {
                  state.ui.submenu['about'] = false
                  state.action('#! close submenu: about', state)
                }
              },
              href          : '/#about'
            }, 'Über uns')
          ]),
          h('span', [
            h('a', {
              className     :  state.ui.submenu['contact'] ?
                s('.item','.item--hover') : s('.item'),
              onmouseenter () {
                if (!state.ui.submenu['contact']) {
                  state.ui.submenu['contact'] = true
                  state.action('#! open submenu: contact', state)
                }
              },
              onmouseleave () {
                if (state.ui.submenu['contact']) {
                  state.ui.submenu['contact'] = false
                  state.action('#! close submenu: contact', state)
                }
              },
              href          : '/#contact'
            }, 'Kontakt')
          ])
        ])
      ])
    )
    // SUBTEMPLATES
    function submenu (state, children) {
      if (!state.ui.submenu) { state.ui.submenu = {} }
      var attributes = {
        className   : s(
          '.submenu', state.ui.submenu['products'] ? '.submenu--visible' : '.submenu--hidden'
        )
      }
      return h('div', attributes,
        state.data.submenu.map(function (entry) {
          return h('span', [
            h('a', {
              className     :  state.ui.submenu[entry.name] ?
                s('.subitem','.subitem--hover') : s('.subitem'),
              onmouseenter () {
                if (!state.ui.submenu[entry.name]) {
                  state.ui.submenu[entry.name] = true
                  state.action('#! open submenu: ' + entry.name, state)
                }
              },
              onmouseleave () {
                if (state.ui.submenu[entry.name]) {
                  state.ui.submenu[entry.name] = false
                  state.action('#! close submenu: ' + entry.name, state)
                }
              },
              href: entry.link
            }, entry.name)
          ])
        })
      )
    }
  }
}
