// _header/header.js
// STACK
var h       = require('virtual-hyperscript-hook')(require('virtual-dom/h'))
var jss     = require('jss')
var css2jss = require('jss-cli/lib/cssToJss')

module.exports = HeaderComponent

function HeaderComponent (state) {
  // STYLING
  // @TODO: use: state.theme.colors
  var sheet  = jss.create().createStyleSheet(css2jss(`
    .header {
      position          : relative;
      background-color  : lightblue;
    }
    .logo {
      display           : inline-block;
    }
    .menubar {
      display           : block;
      position          : absolute;
      bottom            : 0px;
      right             : 20%;
      font-size         : 25px;
      background-color  : lightgreen;
    }
    .menuitem {

      background-color  : orange;
      display           : inline-block;
      padding           : 20px;
    }
    .submenuitem {
      padding           : 10px;
      background-color  : yellow;
    }
    .submenu {
      margin-top        : 20px;
      position          : absolute;
      left              : 0px;
      font-size         : 25px;
      background-color  : lightgreen;
    }
    .isVisible {
      display           : block;
    }
    .isHidden {
      display           : none;
    }
  `),{ named: true, link: true }).attach()
  function s (name, modifier) {
    modifier = modifier ? '  ' + sheet.classes[modifier] : ''
    name     = sheet.classes[name]
    return name + modifier
  }
  // var color = sheet.getRule('.header').prop('background-color')
  // var color = sheet.getRule('.header').prop('background-color', 'green')

  return function Header (state) {
    function submenu (children) {
      if (!state.ui.submenu) { state.ui.submenu = {} }
      var attributes = {
        className   : s('.submenu', state.ui.submenu.products ? '.isVisible' : '.isHidden')
      }
      return h('div', attributes,
        children.map(function (entry) {
          return h('div', { className: s('.submenuitem') }, [
            h('a', { href: entry.link }, entry.name)
          ])
        })
      )
    }
    return h('div', { className: s('.header') }, [
      h('div', { className: s('.logo') }, [
        h('div', h('a', { href: '/'}, [
          h('img', { src: 'redlab.svg', height: '200' })
        ])),
        h('div', {
          className: s('.menubar')
        }, [
          h('span', {
            className: s('.menuitem'),
            onmouseenter: function (event) {
              if (!state.ui.submenu.products) {
                state.ui.submenu.products = true
                state.action('#! open submenu', state)
              }
            },
            onmouseleave: function (event) {
              if (state.ui.submenu.products) {
                state.ui.submenu.products = false
                state.action('#! close submenu', state)
              }
            },
            href: '/#products'
          }, [
            h('a', {
              href: '/#products'
            }, 'Produkte'),
            submenu([{
              link  : '/#products/learning-experience',
              name  : 'Learning Experience'
            },{
              link  : '/#products/female-leadership',
              name  : 'Female Leadership'
            },{
              link  : '/#products/book',
              name  : 'Buch'
            }])
          ]),
          h('span', [
            h('a', {className: s('.menuitem'), href: '/#about'}, 'Über uns')
          ]),
          h('span', [
            h('a', {className: s('.menuitem'), href: '/#contact'}, 'Kontakt')
          ])
        ]),
      ]),
    ])
  }
}
