// _page_home/home.js
// STACK
var h       = require('virtual-hyperscript-hook')(require('virtual-dom/h'))
// CUSTOM
var styling = require('_styling')
var Slider  = require('_slider')
var s
/*
@TODO: BUCH SEITE
-> email subscription (wookmark?)
hier mehr infos zu unserem program - willste das wir uns bei dir melden? (will feedback haben)
*/
module.exports = HomeComponent

function HomeComponent (state) {
  s = s ? s : styling(`
    .content {
      display           : flex;
      flex-direction    : column;
    }
    .banner {
      background-color  : ${state.theme.colors.red};
      justify-content   : center;
      display           : flex;
      flex-direction    : row;
      width             : 100%;
    }
    .boxes {
      align-items       : stretch;
      background-color  : ${state.theme.colors.white};
      display           : flex;
      flex-direction    : row;
      justify-content   : space-around;
    }
    .box {
      background-color  : ${state.theme.colors.cyangrey};
      display           : flex;
      flex-direction    : column;
      font-family       : ${state.theme.font.hnul};
      font-size         : 20px;
      justify-content   : flex-start;
      margin            : 20px;
      padding           : 20px;
      text-align        : center;
      width             : 30%;
    }
    .tag {
      font-weight       : 900;
    }
  `)
  // @TODO: S1.png layout design specification
  var hSlider = Slider(state)
  return function Home (state) {
    return h('div', { className: s('.content') }, [
      h('div', { className: s('.banner') }, hSlider('_page_main', state)),
      h('div', { className: s('.boxes') }, [
        h('div', { className: s('.box') }, [
          h('h3', 'Innovation'),
          h('div', [
            h('p', [
              h('span', 'Alle sprechen darüber: '),
              h('span', { className: s('.tag') }, '#DigitalerWandel '),
              h('span', { className: s('.tag') }, '#entrepreneurialMindeset '),
              h('span', { className: s('.tag') }, '#Innovation '),
              h('span', { className: s('.tag') }, '#Startups'),
              h('span', '. Wir verwandeln Buzzwords in '),
              h('b', 'greifbare Erfahrungen'),
              h('span', '. Wer Neuland erobern will, muss losziehen, eintauchen und erleben. Formate wie die '),
              h('b', 'Innovation Journey'),
              h('span', ' oder der '),
              h('b', 'Innovation Deep Dive'),
              h('span', ` bringen Sie auf Augenhöhe mit den Disruptern und Changern von heute & morgen. Im direkten Austausch gewinnen Sie wertvolle Impulse um Veränderungsprozesse anzustoßen und neue
              `),
              h('b', 'Lösungsansätze'),
              h('span', ' zu entwickeln. ')
            ]),
            h('a', { href: '#innovation' }, 'Mehr...')
          ])
        ]),
        h('div', { className: s('.box') }, [
          h('h3', 'Transformation'),
          h('p', [
            h('span', 'Innovation ist Kultur, keine Abteilung. Der '),
            h('b', 'digitale Wandel'),
            h('span', ` zwingt Organisationen sich schnell zu verändern. Transformation bedeutet unbekannte Prozesse zu probieren, zu justieren und zu implementieren. Wie kann radikale Kreativität,  unternehmerisches Denken und agiles Handeln fruchtbar gefördert werden? Die `),
            h('b', 'Learning Experience'),
            h('span', ` ist eine individuell kuratierte Lernreise zu den Hotspots der Welt. Inspiration und Best Practices aus erster Hand erleben, um das digitale Morgen zu meistern. Schaffen Sie `),
            h('b', 'Synergien'),
            h('span', `, die nachhaltige Wertschöpfung fördern.`),
          ]),
          h('a', { href: '#transformation' }, 'Mehr...')
        ]),
        h('div', { className: s('.box') }, [
          h('h3', 'Leadership'),
          h('p', [
            h('span', `Die Digitalisierung stellt viele Führungskräfte vor völlig neue Herausforderungen. Die Zusammenarbeit verändert. Gefragt sind kollaborative Ansätze, die eine hohe emotionale Intelligenz erfordern. Unsere Leadership Programme unterstützen gezielt Frauen bei der Entwicklung ihrer Führungskarriere. Das Digital Leadership Program bereitet Führungskräfte auf die neuen Anforderungen des `),
            h('b', 'digitalen Wandels'),
            h('span', ' vor. Wir bieten '),
            h('b', 'Strategien, Konzepte und Tools'),
            h('span', ' für individuelles Wachstum.'),
          ]),
          h('a', { href: '#leadership' }, 'Mehr...')
        ])
      ])
    ])
  }
}

/*
  @TODO: Get in touch Button in rot einbauen

  the red lab – we enable growth

  VORSICHT PERSONEN IN ALPHABETISCHER REIHENFOLGE und TITEL VERÄNDERN FEHLER IM MOCK UP
    3. Bianca Praetorius – Co Founder & Partner
    2. Christine Kirbach – Co Founder & Partner
    1. Stefanie Hoffmann – Co Founder & Partner

    1. Stefanie Hoffmann – Co Founder & Partner
    2. Christine Kirbach – Co Founder & Partner
    3. Bianca Praetorius – Co Founder & Partner

Durch fein kuratierte <b>„Experiences“</b> machen wir Innovation und neue Themen für Unternehmen - sowie den Menschen darin - erlebbar. Im Fokus steht dabei immer: Die Implementierung des <b>„Neuen“</b> in die Praxis. Wir unterstützen Unternehmen und Führungskräfte in ihrem individuellen Transformationsprozess. Wir bilden starke <b>Netzwerke</b>, die nachhaltige Wertschöpfung schaffen.

Facebook: https://www.facebook.com/TheRedLab?fref=ts - 373986972623193
Instagram:
Twitter: https://twitter.com/redlabtweets
*/
