// _footer/footer.js
// STACK
var h       = require('virtual-hyperscript-hook')(require('virtual-dom/h'))
// CUSTOM
var styling = require('_styling')
var s
// @TODO: CREDITS:
/*
  Author  : Kippertie
  Webpage : http://www.kippertie.com
  Licence : http://creativecommons.org/licenses/by/3.0/us/
*/
// => fairydust.agency
module.exports = FooterComponent

function FooterComponent (state) {
  s = s ? s : styling(`
    .footer {
      background-color  : ${state.theme.colors.darkergrey};
      display           : flex;
      flex-direction    : column;
      align-items       : center;
      padding-bottom    : 10px;
      /*
      height            : 330px;
      */
    }
    .footercontent {
      background-color  : ${state.theme.colors.darkergrey};
      display           : flex;
      flex-direction    : row;
      align-items       : stretch;
      justify-content   : space-between;
      width             : 940px;
      padding-top       : 40px;
    }

    .navigation {
      display           : flex;
      flex-direction    : column;
    }
    .impressum {
      display           : flex;
      font-size         : ${state.theme.font.avenir.medium.size+'pt'};
      font-family       : ${state.theme.font.avenir.medium.family};
      line-height       : ${state.theme.font.avenir.big.size + 'pt'};
      color             : ${state.theme.colors.darkgrey};
    }
    .link {
      font-size         : ${state.theme.font.avenir.medium.size+'pt'};
      font-family       : ${state.theme.font.avenir.medium.family};
      text-decoration   : none;
      line-height       : ${state.theme.font.avenir.big.size + 'pt'};
      text-transform    : uppercase;
      color             : ${state.theme.colors.bluegrey};
    }
    .link--hover {
    color             : ${state.theme.colors.red};
    }
    .followus {
      display           : flex;
      flex-direction    : column;
    }
    .social {
      display           : flex;
      flex-direction    : column;
      align-items       : flex-end;
      margin-bottom    : 25px;
    }
    .news {
      display           : flex;
      flex-direction    : column;
      align-items       : flex-end;
    }
    .sitemap {
      font-size         : ${state.theme.font.avenir.small.size+'pt'};
      font-family       : ${state.theme.font.avenir.medium.family};
      color             : ${state.theme.colors.darkgrey};
      display           : flex;
      flex-direction    : row;
      justify-content   : flex-end;
      width             : 940px;
      margin-top        : 20px;
      padding-top       : 10px;
      border-top-style  : solid;
      border-top-width  : 1px;
      border-top-color  : ${state.theme.colors.darkgrey};
    }
    .footerlink         {
      padding-right     : 50px;
      text-decoration   : none;
      color             : ${state.theme.colors.darkgrey};
    }
    .footerlink--hover  {
      color             : ${state.theme.colors.white};
    }
  `)
  var hNewsletter = require('_newsletter')(state)
  var hSocialmedia = require('_socialmedia')(state)
  return function Footer (state) {
    return h('div', { className: s('.footer') }, [
      h('div', { className: s('.footercontent') }, [
        h('div', { className: s('.navigation') }, [
          link({
             state     : state,
             namespace : 'footernavigation',
             name      : 'home',
             text      : 'home',
             url       : '/',
             csshover  : '.link--hover',
             cssnormal : '.link'
           }),
          state.data.submenu.filter(function (entry) {
            return entry.name.indexOf('Leadership') === -1
          }).map(function (entry) {
           return link({
              state     : state,
              namespace : 'footernavigation',
              name      : entry.name,
              text      : entry.name,
              url       : entry.link,
              csshover  : '.link--hover',
              cssnormal : '.link'
            })
          }),
          link({
             state     : state,
             namespace : 'footernavigation',
             name      : 'leadership',
             text      : 'Leadership',
             url       : '/#products/leadership',
             csshover  : '.link--hover',
             cssnormal : '.link'
           }),
           link({
              state     : state,
              namespace : 'footernavigation',
              name      : 'about',
              text      : 'Über uns',
              url       : '/#about',
              csshover  : '.link--hover',
              cssnormal : '.link'
            }),
            link({
               state     : state,
               namespace : 'footernavigation',
               name      : 'contact',
               text      : 'Kontakt & Impressum',
               url       : '/#contact',
               csshover  : '.link--hover',
               cssnormal : '.link'
             })
        ]),
        h('div', { className: s('.impressum') }, [
          'read the red lab UG', h('br'),
          'Schlegelstr. 30', h('br'),
          '10115 Berlin', h('br'),
          'Telefon: 0170 7600 280', h('br'),
          'E-Mail: kontakt@theredlab.de', h('br'),
          'www.theredlab.de', h('br')
        ]),
        h('div', { className: s('.followus') }, [
          h('div', { className: s('.social') }, hSocialmedia(state)),
          h('div', { className: s('.news') }, hNewsletter(state))
        ])
      ]),
      h('div', { className: s('.sitemap') }, [
        link({
          state     : state,
          namespace : 'sitemap',
          name      : 'privacy',
          text      : 'Privacy Policy',
          url       : '/#privacy',
          csshover  : '.footerlink--hover',
          cssnormal : '.footerlink'
        }),
        link({
          state     : state,
          namespace : 'sitemap',
          name      : 'terms',
          text      : 'Terms and Conditions',
          url       : '/#terms',
          csshover  : '.footerlink--hover',
          cssnormal : '.footerlink'
        }),
        '© the red lab UG (haftungsbeschränkt)'
      ])
    ])
  }
  // SUBTEMPLATES
  function link ({state, namespace, name, text, url, csshover, cssnormal} = {}) {
    if (!state.ui[namespace]) { state.ui[namespace] = {} }
    return h('a', {
      className     : state.ui[namespace][name] ?
        s(cssnormal,csshover) : s(cssnormal),
      onmouseenter (event) {
        if (!state.ui[namespace][name]) {
          state.ui[namespace][name] = true
          state.action('#! hover: '+name, state)
        }
      },
      onmouseleave (event) {
        if (state.ui[namespace][name]) {
          state.ui[namespace][name] = false
          state.action('#! unhover: '+name, state)
        }
      },
      href          : url
    }, text)
  }
}
