// _footer/footer.js
// STACK
var h       = require('virtual-hyperscript-hook')(require('virtual-dom/h'))
// CUSTOM
var styling = require('_styling')
var s
// @TODO: CREDITS:
/*
  Author  : Kippertie
  Webpage : http://www.kippertie.com
  Licence : http://creativecommons.org/licenses/by/3.0/us/
*/
// => fairydust.agency
module.exports = FooterComponent

function FooterComponent (state) {
  s = s ? s : styling(`
    .footer {
      background-color  : black;
      display           : flex;
      align-items       : stretch;
      justify-content   : space-between;
      padding           : 20px;
    }
    .navigation {
      display           : inline-block;
    }
    .impressum {
      display           : inline-block;
      color             : white;
    }
    .link {
      display           : block;
      padding           : 5px;
      font-size         : 20px;
      text-decoration   : none;
      width             : 200px;
      color             : white;
    }
    .followus {
      display           : flex;
      flex-direction    : column;
      justify-content   : space-between;
    }
    .news {
      color             : white;
    }
    .social {
      padding-bottom    : 20px;
    }
  `)
  var hNewsletter = require('_newsletter')(state)
  var hSocialmedia = require('_socialmedia')(state)
  return function Footer (state) {
    return h('div', { className: s('.footer') }, [
      h('div', { className: s('.navigation') }, [
        h('a', { className: s('.link'), href: '/'}, 'Home'),
        h('a', { className: s('.link'), href: '/#products/learning-experience'}, 'Learning Experience'),
        h('a', { className: s('.link'), href: '/#products/female-leadership'}, 'Female Leadership'),
        h('a', { className: s('.link'), href: '/#products/book'}, 'Buch'),
        h('a', { className: s('.link'), href: '/#about'}, 'Über uns'),
        h('a', { className: s('.link'), href: '/#contact'}, 'Kontakt & Impressum')
      ]),
      h('div', { className: s('.impressum') }, [
        'read the red lab UG', h('br'),
        'Schlegelstr. 30', h('br'),
        '10115 Berlin', h('br'),
        'Telefon: 0170 7600 280', h('br'),
        'E-Mail: kontakt@theredlab.de', h('br'),
        'www.theredlab.de', h('br')
      ]),
      h('div', { className: s('.followus') }, [
        h('div', { className: s('.news') }, hNewsletter(state)),
        h('div', { className: s('.social') }, hSocialmedia(state))
      ])
    ])
  }
}
