// _content/content.js
// STACK
var h       = require('virtual-hyperscript-hook')(require('virtual-dom/h'))
var jss     = require('jss')
var css2jss = require('jss-cli/lib/cssToJss')
// CUSTOM
var Banner  = require('_banner')

module.exports = ContentComponent

function ContentComponent (state) {
  // STYLING
  // @TODO: use: state.theme.colors
  var hBanner   = Banner(state)
  var sheet  = jss.create().createStyleSheet(css2jss(`
    .content {
      background-color  : ${state.theme.colors.white};
      display           : flex;
      align-items       : center;
      justify-content   : space-around;
      border-radius     : 4px;
      height            : 300px;
    }
    .box {
      background-color  : ${state.theme.colors.cyangrey};
      text-align        : center;
      padding           : 20px;
      font-size         : 20px;
      height            : 200px;
      width             : 300px;
    }
  `),{ named: true, link: true }).attach()
  function s (name, modifier) {
    modifier = modifier ? '  ' + sheet.classes[modifier] : ''
    name     = sheet.classes[name]
    return name + modifier
  }
  // var color = sheet.getRule('.header').prop('background-color')
  // var color = sheet.getRule('.header').prop('background-color', 'green')
  return function Content (state) {
    // @TODO: maybe use another router :-)
    return h('div', [
      h('div', hBanner(state)),
      h('div', { className: s('.content') }, [
        h('div', { className: s('.box') }, [
          h('div', 'Learning Experience'),
          h('div', ' ... ')
        ]),
        h('div', { className: s('.box') }, [
          h('div', 'Female Leadership'),
          h('div', ' ... ')
        ]),
        h('div', { className: s('.box') }, [
          h('div', 'Buch'),
          h('div', ' ... ')
        ])
      ])
    ])
  }
}
/*
Social Media Feed:

- boxen analog zu boxen

- facebook, instagram, twitter, blog
- https://twitter.com/the_red_lab
- https://www.facebook.com/TheRedLab?fref=ts
- INSTGRAM ACCOUNT
- BLOG ACCOUNT
=> general electrics feed

// router.addRoute('/students', function (m) {
//   return h('div', m.data.feed.map(function (s) {
//     var name = s.key.split('!')[1]
//     return h('div.student',
//       h('a', { href: '/feed/' + name }, name)
//     )
//   }))
// })
// router.addRoute('/student/:name', function (m) {
//   var house = sorthash(m.params.name)
//   return h('div', [
//     h('h1', m.params.name),
//     h('div', 'a student of house ' + house)
//   ])
// })

BUCH SEITE
-> email subscription (wookmark?)
hier mehr infos zu unserem program - willste das wir uns bei dir melden? (will feedback haben)
*/
