// _content/content.js
// STACK
var h       = require('virtual-hyperscript-hook')(require('virtual-dom/h'))
// CUSTOM
var styling = require('_styling')
var Banner  = require('_banner')
var s
/*
@TODO: BUCH SEITE
-> email subscription (wookmark?)
hier mehr infos zu unserem program - willste das wir uns bei dir melden? (will feedback haben)
*/
module.exports = ContentComponent

function ContentComponent (state) {
  s = s ? s : styling(`
    .content {
      align-items       : center;
      background-color  : ${state.theme.colors.white};
      display           : flex;
      justify-content   : space-around;
    }
    .box {
      background-color  : ${state.theme.colors.cyangrey};
      font-family       : ${state.theme.font.hnul};
      font-size         : 20px;
      margin            : 20px;
      min-height        : 300px;
      min-width         : 200px;
      padding           : 20px;
      text-align        : center;
      width             : 25%;
    }
  `)
  var hBanner = Banner(state)
  return function Content (state) {
    return h('div', [
      h('div', hBanner(state)),
      h('div', { className: s('.content') }, [
        h('div', { className: s('.box') }, [
          h('h3', 'Learning Experience'),
          h('div', ' lorem ipsum ')
        ]),
        h('div', { className: s('.box') }, [
          h('h3', 'Female Leadership'),
          h('div', ' lorem ipsum ')
        ]),
        h('div', { className: s('.box') }, [
          h('h3', 'Buch'),
          h('div', ' lorem ipsum ')
        ])
      ])
    ])
  }
}
