// _banner/banner.js
// STACK
var h       = require('virtual-hyperscript-hook')(require('virtual-dom/h'))
var jss     = require('jss')
var css2jss = require('jss-cli/lib/cssToJss')

module.exports = BannerComponent

function BannerComponent (state) {
  // STYLING
  // @TODO: use: state.theme.colors
  var sheet  = jss.create().createStyleSheet(css2jss(`
    .banner {
      align-items       : center;
      background-color  : ${state.theme.colors.white};
      display           : flex;
      justify-content   : center;
      width             : 100%;
    }
  `),{ named: true, link: true }).attach()
  function s (name, modifier) {
    modifier = modifier ? '  ' + sheet.classes[modifier] : ''
    name     = sheet.classes[name]
    return name + modifier
  }
  // var color = sheet.getRule('.header').prop('background-color')
  // var color = sheet.getRule('.header').prop('background-color', 'green')

  return function Banner (state) {
    return ( // @TODO: Banner 3 bilder rotation
      h('video', {
      className: s('.banner'),
      autoplay: true,
      loop: true,
      muted: true,
      src: 'assets/bannervideo.mp4'
      })
    )
  }
}
