// _socialmedia/socialmedia.js
// STACK
var h       = require('virtual-hyperscript-hook')(require('virtual-dom/h'))
// CUSTOM
var styling = require('_styling')
var s
var isretina = require('is-retina')

module.exports = SocialmediaComponent

/*
Facebook: https://www.facebook.com/TheRedLab?fref=ts - 373986972623193
Instagram:
Twitter: https://twitter.com/redlabtweets
*/
function SocialmediaComponent (state) {
  s = s ? s : styling(`
    .socialmedia {
      display: flex;
      flex-direction: row;
      justify-content: space-between;
      width: 100%;
    }
    ${
      ['email', 'facebook', 'twitter', 'linkedin', 'xing', 'instagram'].map(function (name) {
        return `
          .${name} {
            display           : flex;
            background-image: url('${
              state.data.socialmedia[name][isretina()?'retina':'normal']
            }');
            margin-left: 10px;
            background-position: 0px 0px;
            background-repeat : no-repeat;
            width             : 42px;
            height            : 42px;
          }
          .${name}--hover {
            display           : flex;
            background-image: url('${
              state.data.socialmedia[name][isretina()?'retina':'normal']
            }');
            background-position: 0px -42px;
            background-repeat : no-repeat;
            width             : 42px;
            height            : 42px;
          }
        `
      }).join('')
    }
  `)
  return function Socialmedia (state) {
    return h('div', { className: s('.socialmedia') }, [
      linksocial({
        state: state,
        namespace: 'socialmedia',
        name: 'email',
        text: '',
        url: '/#email',
        csshover: '.email--hover',
        cssnormal: '.email',
        target: '_blank'
      }),
      linksocial({
        state: state,
        namespace: 'socialmedia',
        name: 'facebook',
        text: '',
        url: '/#facebook',
        csshover: '.facebook--hover',
        cssnormal: '.facebook',
        target: '_blank'
      }),
      linksocial({
        state: state,
        namespace: 'socialmedia',
        name: 'twitter',
        text: '',
        url: '/#twitter',
        csshover: '.twitter--hover',
        cssnormal: '.twitter',
        target: '_blank'
      }),
      linksocial({
        state: state,
        namespace: 'socialmedia',
        name: 'linkedin',
        text: '',
        url: '/#linkedin',
        csshover: '.linkedin--hover',
        cssnormal: '.linkedin',
        target: '_blank'
      }),
      linksocial({
        state: state,
        namespace: 'socialmedia',
        name: 'xing',
        text: '',
        url: '/#xing',
        csshover: '.xing--hover',
        cssnormal: '.xing',
        target: '_blank'
      }),
      linksocial({
        state: state,
        namespace: 'socialmedia',
        name: 'instagram',
        text: '',
        url: '/#instagram',
        csshover: '.instagram--hover',
        cssnormal: '.instagram',
        target: '_blank'
      })
    ])
  }
}
// SUBTEMPLATES
function linksocial ({state, namespace, name, text, url, csshover, cssnormal, target} = {}) {
  if (!state.ui[namespace]) { state.ui[namespace] = {} }
  return h('a', {
    className     : state.ui[namespace][name] ?
      s(cssnormal,csshover) : s(cssnormal),
    onmouseenter (event) {
      if (!state.ui[namespace][name]) {
        state.ui[namespace][name] = true
        state.action('#! hover: '+name, state)
      }
    },
    onmouseleave (event) {
      if (state.ui[namespace][name]) {
        state.ui[namespace][name] = false
        state.action('#! unhover: '+name, state)
      }
    },
    href          : url,
    target        : target
  }, text)
}
